src/components/device/ha-device-trigger-picker.ts

```diff
@@ -0,0 +1,148 @@
+  @property() public presetTrigger?: DeviceTrigger;
+  private noTrigger: DeviceTrigger = {
+    device_id: this.deviceId || "",
+    platform: "device",
+  };
+
+  private _sortedTriggers = memoizeOne((triggers?: DeviceTrigger[]) => {
+    return triggers || [];
+  });
+
+  protected render(): TemplateResult | void {
+    const noTriggers = this._sortedTriggers(this.triggers).length === 0;
+    return html`
+      <paper-dropdown-menu-light .label=${this.label} ?disabled=${noTriggers}>
+        <paper-listbox
+          slot="dropdown-content"
+          .selected=${this._trigger}
+          attr-for-selected="trigger"
+          @iron-select=${this._triggerChanged}
+          id="listbox"
+        >
+          ${noTriggers
+            ? html`
+                <paper-item .trigger=${this.noTrigger}>
+                  No triggers
+                </paper-item>
+              `
+            : ""}
+          ${this._sortedTriggers(this.triggers).map(
+            (trigger) => html`
+              <paper-item .trigger=${trigger}>
+                ${this.hass!.localize(
+                  `ui.panel.config.automation.editor.triggers.type.device.trigger_type.${
+                    trigger.type
+                  }`,
+                  "entity_id",
+                  trigger.entity_id
+                    ? computeStateName(this.hass!.states[trigger.entity_id])
+                    : "",
+                  "event",
+                  trigger.event
+                )}
+              </paper-item>
+            `
+          )}
+        </paper-listbox>
+      </paper-dropdown-menu-light>
+    `;
+  }
+
+  private get _trigger() {
+    return this.trigger;
+  }
+
+  protected updated(changedProps) {
+    super.updated(changedProps);
+
+    // Reset trigger if device-id has changed
+    if (changedProps.has("deviceId")) {
+      this.noTrigger = { device_id: this.deviceId || "", platform: "device" };
+      if (this.deviceId) {
+        fetchDeviceTriggers(this.hass!, this.deviceId).then((trigger) => {
+          this.triggers = trigger.triggers;
+          if (this.triggers.length > 0) {
+            // Set first trigger as default
+            this.trigger = this.triggers[0];
+          } else if (triggersEqual(this.noTrigger, this.presetTrigger)) {
+            this.trigger = this.noTrigger;
+          }
+          for (const trig of this.triggers) {
+            // Try to find a trigger matching existing trigger loaded from stored automation
+            if (triggersEqual(trig, this.presetTrigger)) {
+              this.trigger = trig;
+            }
+          }
+        });
+      } else {
+        // No device, clear the list of triggers
+        this.triggers = [];
+        this.trigger = this.noTrigger;
+      }
+    }
+
+    // The triggers property has changed, force the listbox to update
+    if (changedProps.has("triggers")) {
+      const listbox = this.shadowRoot!.getElementById("listbox") as any;
```

Comment:
Since you only have one listbox, you can also select by tag name. All Polymer elements have TS types, so you should be able to map it the right type.

Reply:
How to do that?
If I change to `const listbox = this.getElementsByTagName("PAPER-LISTBOX") as any;` I just get an empty array.

Reply:
`paper-listbox`

Reply:
`const listbox = this.getElementsByTagName("paper-listbox") as any;` also returns an empty array

Reply:
`render()` renders into our shadowRoot, so do `this.shadowRoot.querySelector('paper-listbox')`

Reply:
Fixed